### DISCUSSION TITLE
Using a model other than GP

### DISCUSSION THREAD

**Discussion by eloiseum** [meta-pytorch/botorch#1263] [Q&A] [answered]
Hi everyone, 

I wanted to try another model instead of GP (e.g. random forests), but I am a bit lost on how to implement it using BoTorch. Has anyone done it or any hints on how to make this change? I have read the documentation but found very little guidance on how to implement other models.

Many thanks in advance!

**Comment by Balandat** [comment, marked as answer]
Hi, how were you planning on addressing predictive uncertainty with your model (e.g. RFs)? 

At the end of the day what you’ll need to make things work is a `Model` class that has a `posterior()` method that upon calling reruns a `Posterior` object. If you don’t have a closed form posterior then you can still use this setup with Monte Carlo acquisition functions so long as you can draw samples from your `Posterior` object. If you want to use gradient based optimization of the acquisition function then on top of that you’ll need to be able to backprop through those samples.
